src/routes/+page.svelte

```diff
@@ -17,13 +29,119 @@
 	let translateX = $derived(percentMoved * ((OVERLAY_OFFSET_LEFT_PERCENT / 100) * innerWidth));
 	let translateY = $derived(percentMoved * (-(OVERLAY_OFFSET_BOTTOM_PERCENT / 100) * innerHeight));
 
+	let sectionRefs: HTMLElement[] = [];
+
+	const intersectionCallback = (entries: IntersectionObserverEntry[]) => {
+		entries.forEach((entry) => {
+			const section = entry.target as HTMLElement;
+
+			// Select all images within the current section
+			const images = section.querySelectorAll<HTMLImageElement>('img');
+
+			if (entry.isIntersecting) {
+				// Show images
+				images.forEach((image) => {
+					image.style.transitionDelay = '0s';
+					image.classList.add('visible');
+				});
+			} else {
+				// Hide images
+				images.forEach((image) => {
+					image.classList.remove('visible');
+				});
+			}
+		});
+	};
+
+	onMount(() => {
+		const observer = new IntersectionObserver(intersectionCallback, {
+			threshold: 0.1 // Trigger when 10% of the section is visible
+		});
+
+		sectionRefs.forEach((section) => observer.observe(section));
+
+		return () => {
+			sectionRefs.forEach((section) => observer.unobserve(section));
+		};
+	});
+
+	let storeSection = (el: HTMLElement) => {
+		if (el && !sectionRefs.includes(el)) {
+			sectionRefs.push(el);
+		}
+	};
+
+	const illustrations: Record<number, SectionIllustrations> = {
+		0: {
+			left: {
+				image: hero0Left,
+				className: 'hero-0-left'
+			},
+			right: {
+				image: hero0Right,
+				className: 'hero-0-right'
+			}
+		},
+		1: {
+			right: {
+				image: hero1Right,
+				className: 'hero-1-right'
+			}
+		},
+		2: {
+			left: {
+				image: hero2Left,
+				className: 'hero-2-left'
+			}
+		},
+		3: {
+			left: {
+				image: hero3Left,
+				className: 'hero-3-left'
+			},
+			right: {
+				image: hero3Right,
+				className: 'hero-3-right'
+			}
+		},
+		4: {
+			left: {
+				image: hero4Left,
+				className: 'hero-4-left'
```

Comment:
ah yep you're totally right! I originally thought I'd keep these around at least temporarily in case there were more tweaks we wanted to make, but as you've said they're dynamically generated, so no need to defined them in the `illustrations` object as well.